### ISSUE TITLE
Redirect to login screen when GraphQL response indicates not being properly authenticated

### ISSUE THREAD

**Issue by vyacheslav-pushkin** [Amplicode/amplicode-frontend#12]
### TL;DR

Redirect the user to the login screen when GraphQL request contains an error object indicating not being properly **authenticated**.

### Details

Currently the user is being redirected to login screen when GraphQL response has status 401.

However, the common practice for GraphQL servers is to return status 200 for all errors coming from GraphQL layer, which includes lack of authentication, expired authentication or lack of authorization (lack of privileges). A non-200 return status often indicates an error coming from outside GraphQL, e.g. from transport layer.

Therefore **in addition to** looking for 401 status we should check the error object in the response.

Since the exact structure of error object will differ from backend to backend, we should:

- default to the format used in `jmix2-petclinic` (standard format coming from Spring)
- scaffold a `TODO` (similar to how we do it in entity list components) and provide instructions for the framework user to modify the code to match the backend error object format

Note that in case when a request fails due to the user not being **authorized** to make it (i.e. user not having sufficient privileges) rather than not being **authenticated** (not logged in, expired token) we **shouldn't** automatically redirect to login screen. Discuss with backend team how to discern between these two cases.
